### PR TITLE
Fix Glass focus interaction position

### LINKED ISSUE #1100: Stale hover position overrides focus-driven Glass interaction placement
## Summary

`GlassInteractionController` never clears its recorded hover position after the pointer exits, so a subsequent keyboard focus (which carries no position) renders its lighting/refraction response at the stale hover location instead of centered.

## Confirmed problem

- `hoverPosition` (`GlassInteractionController.kt:194`) is set whenever the pointer hovers inside bounds (`:306`, inside `onPointerEvent`) but is never reset to `null`.
- Neither the `PointerEventType.Exit` branch (`:309-311`, which sets `rawHovered = false`) nor `cancelPointerInput` (`:336-341`, same) clears `hoverPosition`.
- `updatePositionForCurrentInputs` (`:432-439`) resolves the render position as:
  ```kotlin
  rawPosition.takeIf { rawPressed } ?: hoverPosition ?: sourcePosition ?: rawPosition ?: size.center
  ```
  `hoverPosition` outranks `sourcePosition` (the focus-driven position), so once a mouse has ever hovered the component, it takes priority over focus indefinitely.

## Reproduction

1. Hover the pointer near an edge of an interactive Glass surface (nothing necessarily visible if the hover response is subtle).
2. Move the mouse away — `rawHovered` becomes false, the hover response fades toward identity, so nothing looks wrong yet.
3. Tab-focus the element (no mouse involved) — the focus glow/refraction renders offset toward the stale hover point instead of the element's center.

## Suggested direction

Clear `hoverPosition` (set to `null`) in both the `PointerEventType.Exit` branch and `cancelPointerInput`, alongside the existing `rawHovered = false` resets.

## Acceptance criteria

- After a hover-then-exit, a subsequent focus-only interaction renders centered (or at the resolved `sourcePosition`), not at the old hover coordinate.
- Add a controller test: hover near an edge, exit, then focus, and assert the render position is not the stale hover position.

### PR DESCRIPTION
**Pull request by chrisbanes**
## Summary

- distinguish active pointer inputs from retained idle hover coordinates
- center focus-driven interactions when no source position is available
- preserve active press/hover precedence and last-position idle behavior
- cover hover-exit-focus, pointer precedence, and cancellation-focus transitions

## Verification

- `./gradlew :haze-glass:jvmTest --tests dev.chrisbanes.haze.glass.GlassInteractionControllerTest :haze-glass:spotlessCheck`
- `./gradlew :haze-glass:allTests :haze-glass:spotlessCheck`
- `git diff --check origin/main...HEAD`
- `/code-review origin/main` — Standards: no findings; Spec: no findings
- `/review-and-simplify-changes` — no findings
- `/ponytail-review` — lean, no findings

## Risk

Low. The change is internal to interaction-position selection and leaves animation specs,
response ownership, and retained idle-position behavior unchanged.

Fixes #1100

### CODE REVIEW

File: haze-glass/src/commonMain/kotlin/dev/chrisbanes/haze/glass/GlassInteractionController.kt [hunk 1]
```diff
@@ -431,11 +431,13 @@ internal class GlassInteractionController(
...
-      ?: hoverPosition
-      ?: sourcePosition
-      ?: rawPosition
-      ?: size.center
+    val target = when {
+      rawPressed -> rawPosition ?: sourcePosition ?: size.center
+      sourcePresses.isNotEmpty() -> sourcePosition ?: rawPosition ?: size.center
+      rawHovered -> hoverPosition ?: size.center
+      sourceHovers.isNotEmpty() || sourceFocuses.isNotEmpty() -> size.center
+      else -> rawPosition ?: hoverPosition ?: size.center
```
[the remark this answers is not in the document]
Comment (chrisbanes) [reply haze-glass/src/commonMain/kotlin/dev/chrisbanes/haze/glass/GlassInteractionController.kt]: Fixed in 6a46d0cc. The idle fallback now prefers the retained hover position over an older press position, while the active press/hover/focus precedence remains unchanged. Added a press → hover/move → hover-exit regression test; it failed against the previous PR head and passes with the fix. Full :haze-glass:allTests and spotlessCheck pass.
